toolkit/tools/imagegen/configuration/parse_partition.go

```diff
@@ -0,0 +1,163 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+// Utility to create and manipulate disks and partitions
+
+package configuration
+
+import (
+	"bufio"
+	"os"
+	"strconv"
+	"strings"
+
+	"microsoft.com/pkggen/internal/logger"
+)
+
+// In kickstart installation, the kickstart file will include a file (/tmp/part-include)
+// that includes the partitioning commands populated by executing the preinstall script.
+// This function aims to parse the partitioning instruction from this kickstart partition file
+// and fill up the required Disk and PartitionSetting information.
+
+// Sample partition configuration for reference: https://www.golinuxhub.com/2018/05/sample-kickstart-partition-example-raid/
+func ParseKickStartPartitionScheme(config *Config, partitionFile string) (err error) {
+
+	var (
+		parseCmd           string
+		partitionFlags     []string
+		partitionTableType PartitionTableType
+		diskInfo           map[string]int
+	)
+
+	// Check whether the config already contains partition schema
```

Comment:
When would this happen?

Reply:
My thoughts are this parser would only be used for kickstart-style unattended installations. And for these users, they only need to specify the preinstall scripts in the image config json and no direct partitioning input is needed, which is the same as what they are doing with the kickstart file. For this user scenario, we shouldn't allow defining partitioning in both image config file and preinstall script. Thus, if the image config file already contains partition data, then this is not considered a kickstart unattended installation and the parser should not be executed.